server/src/main/java/org/opensearch/search/query/QueryCollectorContext.java

```diff
@@ -198,16 +203,33 @@ protected InternalProfileCollector createWithProfiler(InternalProfileCollector i
 
                 for (CollectorManager<? extends Collector, ReduceableSearchResult> manager : subs) {
                     final Collector collector = manager.newCollector();
-                    if (!(collector instanceof InternalProfileCollector)) {
-                        throw new IllegalArgumentException("non-profiling collector");
-                    }
-                    subCollectors.add((InternalProfileCollector) collector);
+                    subCollectors.add(new InternalProfileCollector(collector, CollectorResult.REASON_AGGREGATION, Collections.emptyList()));
                 }
 
                 final Collector collector = MultiCollector.wrap(subCollectors);
                 return new InternalProfileCollector(collector, REASON_SEARCH_MULTI, subCollectors);
             }
 
+            @Override
+            protected InternalProfileCollectorManager createWithProfiler(InternalProfileCollectorManager in) {
+                final List<CollectorManager<?, ReduceableSearchResult>> managers = new ArrayList<>();
+                final List<InternalProfileCollectorManager> children = new ArrayList<>();
+                managers.add(in);
+                children.add(in);
+                for (CollectorManager<? extends Collector, ReduceableSearchResult> manager : subs) {
+                    InternalProfileCollectorManager subCollectorManager = new InternalProfileCollectorManager(
+                        manager,
+                        CollectorResult.REASON_AGGREGATION,
+                        Collections.emptyList()
+                    );
```

Comment:
Lets change this same way as above:

```
for (CollectorManager<? extends Collector, ReduceableSearchResult> manager : subs) {
    final InternalProfileCollectorManager subCollectorManager;
    if (manager instanceof AggregationCollectorManager) {
        subCollectorManager = new InternalProfileCollectorManager(manager,
                            ((AggregationCollectorManager) manager).getCollectorReason(),
                            Collections.emptyList());
    } else {
         subCollectorManager = new InternalProfileCollectorManager(manager,
                            REASON_SEARCH_MULTI, Collections.emptyList());
    }
    managers.add(subCollectorManager);
    children.add(subCollectorManager);
}
```

Reply:
Is it actually right? I think we don't need to wrap managers at all and use `in` as the entry point (it is already `InternalProfileCollectorManager`):

```
            @Override
            protected InternalProfileCollectorManager createWithProfiler(InternalProfileCollectorManager in) {
                final List<InternalProfileCollectorManager> children = new ArrayList<>();

                for (CollectorManager<? extends Collector, ReduceableSearchResult> manager : subs) {
                    InternalProfileCollectorManager subCollectorManager = new InternalProfileCollectorManager(
                        manager,
                        CollectorResult.REASON_AGGREGATION,
                        Collections.emptyList()
                    );
                    children.add(subCollectorManager);
                }

                return new InternalProfileCollectorManager(in, REASON_SEARCH_MULTI, children);
            }
```